test/ruby/test_defined.rb

```diff
@@ -303,6 +303,20 @@ def x; b {return defined?(super)}; end
     assert_equal("super", o.x, bug8367)
   end
 
+  def test_super_in_basic_object
+    BasicObject.class_eval do
+      def a
+        defined?(super)
+      end
+    end
+
+    assert_nil(a)
+
+    BasicObject.class_eval do
```

Comment:
Let’s make sure this is called in an ensure block in case assert_nil (or other code) raises an exception.

Reply:
great idea! https://github.com/ruby/ruby/pull/7758/commits/7c78de578e0f1c0f1a16f9610a32e18fa23a7a25